app/services/motion_service.rb

```diff
@@ -23,19 +23,25 @@ def self.update(motion:, params:, actor:)
       sync_search_vector = motion.name_changed? || motion.description_changed?
 
       if motion.name_changed?
-        Events::MotionNameEdited.publish!(motion, actor)
+        event = Events::MotionNameEdited.publish!(motion, actor)
       end
 
       if motion.description_changed?
-        Events::MotionDescriptionEdited.publish!(motion, actor)
+        event = Events::MotionDescriptionEdited.publish!(motion, actor)
       end
 
       if motion.closing_at_changed?
-        Events::MotionCloseDateEdited.publish!(motion, actor)
+        event = Events::MotionCloseDateEdited.publish!(motion, actor)
       end
 
       motion.save
       ThreadSearchService.index! motion.discussion_id if sync_search_vector
+      if event
```

Comment:
This is super confusing. I think a better way to handle it is returning an event if it exists, and nil otherwise (then the restful controller will default to responding with the collection)

```
return event.reload if event
```

Also why do we need to reload?